### ISSUE TITLE
Don't use ``pkg_resources.declare_namespace(__name__)``

### ISSUE THREAD

**Issue by DanielNoord** [googleads/google-ads-python#766]
**Describe the bug:**

Importing after building the package with `setuptools` 67.3 and later will raise a `DeprecationWarning` because of the deprecation of `pkg_resources.declare_namespace`: https://setuptools.pypa.io/en/latest/history.html#v67-3-0

**Steps to Reproduce:**

**Expected behavior:**

Remove this line from ``__init__.py`` and use the method as described in PEP 420.

**Comment by BenRKarl** [comment]
@DanielNoord thank you for pointing this out, we'll get it updated in a future release.

**Comment by BenRKarl** [comment]
@DanielNoord I'm having trouble reproducing the deprecation warning, could you add some details as to how you're able to get it to appear? What I've tried so far is installing `setuptools==67.3.1` and installing the library into a clean virtual environment, both from my local repository and from PyPI.

**Comment by BenRKarl** [comment]
Yes, same result when using pip version `23.1.1`

Which version of Python are you using?

**Comment by DanielNoord** [comment]
I could reproduce on 3.10.

**Comment by DanielNoord** [comment]
https://discuss.python.org/t/announcement-pip-23-1-release/25844/6

This might be related. Can you reproduce with `23.1.2`?

**Comment by BenRKarl** [comment]
@DanielNoord I'm still unable to reproduce with Python `3.10.5` and `pip==23.1.2`...

It would be useful to get more specifics about your environment. When you have a moment, could you run the below commands in order, capturing the output that shows the deprecation warning?

```
python -V
python -m venv <path to venv>
source <path to venv>/bin/activate
python -m pip list
python -m pip install -U pip setuptools
python -m pip list
cd to google-ads-python
python -m pip install .
python -m pip list
```

**Comment by DanielNoord** [comment]
I could reproduce with:
```console
python3.11 -m virtualenv .venv
source .venv/bin/activate
python -m pip install pip==23.1
python -m pip install setuptools==67.3.1
python -m pip install google-ads
python -m pip install pytest
vim test.py
python -m pytest test.py
```

```console
❯ cat test.py
from google.ads.googleads.client import GoogleAdsClient

def test():
    assert False
```

Output:
```console
❯ python -m pytest test.py
============================= test session starts ==============================
platform linux -- Python 3.11.3, pytest-7.3.1, pluggy-1.0.0
rootdir: /tmp
collected 1 item                                                               

test.py F                                                                [100%]

=================================== FAILURES ===================================
_____________________________________ test _____________________________________

    def test():
>       assert False
E       assert False

test.py:4: AssertionError
=============================== warnings summary ===============================
.venv/lib/python3.11/site-packages/pkg_resources/__init__.py:2803
.venv/lib/python3.11/site-packages/pkg_resources/__init__.py:2803
.venv/lib/python3.11/site-packages/pkg_resources/__init__.py:2803
.venv/lib/python3.11/site-packages/pkg_resources/__init__.py:2803
.venv/lib/python3.11/site-packages/pkg_resources/__init__.py:2803
.venv/lib/python3.11/site-packages/pkg_resources/__init__.py:2803
  /tmp/.venv/lib/python3.11/site-packages/pkg_resources/__init__.py:2803: DeprecationWarning: Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
    declare_namespace(pkg)

.venv/lib/python3.11/site-packages/pkg_resources/__init__.py:2308
.venv/lib/python3.11/site-packages/pkg_resources/__init__.py:2308
.venv/lib/python3.11/site-packages/pkg_resources/__init__.py:2308
  /tmp/.venv/lib/python3.11/site-packages/pkg_resources/__init__.py:2308: DeprecationWarning: Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
    declare_namespace(parent)

.venv/lib/python3.11/site-packages/google/rpc/__init__.py:20
  /tmp/.venv/lib/python3.11/site-packages/google/rpc/__init__.py:20: DeprecationWarning: Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
    pkg_resources.declare_namespace(__name__)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED test.py::test - assert False
======================== 1 failed, 10 warnings in 0.14s ========================
```

Running with ``python -m pytest test.py -Werror` shows that the errors originate from imports in this package.

**Comment by BenRKarl** [comment]
Thanks @DanielNoord I was able to reproduce the warning using those steps. What I found was that if I remove the `declare_namespace` statements from our library, deprecation warnings get raised from other external libraries such as `grpcio`. I'm going to coordinate internally to see if there can be a shared approach to this, since we share the same namespace as other Google libraries. I'll try to post an update here as soon as possible.
